framework/src/modules/nft/endpoint.ts

```diff
@@ -111,18 +111,28 @@ export class NFTEndpoint extends BaseEndpoint {
 
 		const userStore = this.stores.get(UserStore);
 		const nftData = await nftStore.get(context.getImmutableMethodContext(), nftID);
-		const userData = await userStore.get(
-			context.getImmutableMethodContext(),
-			userStore.getKey(nftData.owner, nftID),
-		);
+		if (nftData.owner.length === LENGTH_ADDRESS) {
+			const userData = await userStore.get(
+				context.getImmutableMethodContext(),
+				userStore.getKey(nftData.owner, nftID),
+			);
+
+			return {
+				owner: nftData.owner.toString('hex'),
+				attributesArray: nftData.attributesArray.map(attribute => ({
+					module: attribute.module,
+					attributes: attribute.attributes.toString('hex'),
+				})),
+				lockingModule: userData.lockingModule,
+			};
+		}
 
 		return {
 			owner: nftData.owner.toString('hex'),
 			attributesArray: nftData.attributesArray.map(attribute => ({
 				module: attribute.module,
 				attributes: attribute.attributes.toString('hex'),
 			})),
-			lockingModule: userData.lockingModule,
 		};
 	}
```

Comment:
Is it possible to optimize this code and do the check `if (nftData.owner.length === LENGTH_ADDRESS)` immediately before returning the `lockingModule`? As it is now, it looks a bit duplicated. Just a minor comment